sessionctx/variable/sysvar.go

```diff
@@ -1380,6 +1380,10 @@ var defaultSysVars = []*SysVar{
 		s.RcReadCheckTS = TiDBOptOn(val)
 		return nil
 	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBRemoveOrderbyInSubquery, Value: BoolToOnOff(DefTiDBRemoveOrderbyInSubquery), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
+		s.RemoveOrderbyInSubquery = TiDBOptOn(val)
+		return nil
+	}},
```

Comment:
I think it's not a global variable in this way. Mark the ScopeGlobal will enable the persistence to GLOBAL_VARIABLES table while the value persisted is never inherited by session (You use DefTiDBRemoveOrderbyInSubquery to init s. RemoveOrderbyInSubquery).

You can just remove ScopeGlobal or implement the full ScopeGlobal logic.

Reply:
Thanks, I have removed `ScopeGlobal` as suggested.

Reply:
I think this looked right to me, but I'm not sure because of the force-push.

The `Value: ` only refers to the default value, once it is persisted it will use what is stored internally. `SetSession` is called both when a session value is changed, and on init for new sessions (on init the val will be the global value, which matches the GLOBAL_VARIABLES -- although we have a cache of it internally called sysvar_cache).

Reply:
The downside of ScopeSession only is that it doesn't persist, so each time a user wants this optimization to apply -- they'll need to `SET sysvar = x`. Which limits its appeal.

Reply:
You are right, my fault, I mixed the situation, @yuanhsh please ignore my previous comment.

Reply:
OK, I will revert to previous version.
@morgo Thank you for advising!